File: infrastructure/spring-boot-r2dbc-persistence-adapter/src/testFixtures/java/codes/fdk/blueprint/api/infrastructure/persistence/r2dbc/PostgresContainerExtension.java
package codes.fdk.blueprint.api.infrastructure.persistence.r2dbc;

import org.junit.jupiter.api.extension.AfterEachCallback;
import org.junit.jupiter.api.extension.BeforeAllCallback;
import org.junit.jupiter.api.extension.ExtensionContext;
import org.testcontainers.containers.PostgreSQLContainer;

public class PostgresContainerExtension implements BeforeAllCallback, AfterEachCallback {

    private static final PostgreSQLContainer<?> postgreSQLContainer;

    static {
        postgreSQLContainer = new PostgreSQLContainer<>("postgres:13-alpine");
        postgreSQLContainer.start();
    }

    @Override
    public void beforeAll(ExtensionContext context) {
        final String r2dbcUrl = String.format(
                "r2dbc:postgresql://%s:%d/%s",
                postgreSQLContainer.getHost(),
                postgreSQLContainer.getFirstMappedPort(),
                postgreSQLContainer.getDatabaseName()
        );

        System.setProperty("spring.r2dbc.url", r2dbcUrl);
        System.setProperty("spring.r2dbc.username", postgreSQLContainer.getUsername());
        System.setProperty("spring.r2dbc.password", postgreSQLContainer.getPassword());
    }

    @Override
    public void afterEach(ExtensionContext context) throws Exception {
        postgreSQLContainer.execInContainer(
                "psql",
                "-U",
                postgreSQLContainer.getUsername(),
                "-d",
                postgreSQLContainer.getDatabaseName(),
                "-c",
                "TRUNCATE TABLE categories;"
        );
    }

}

File: infrastructure/vertx-postgres-persistence-adapter/src/test/kotlin/codes/fdk/blueprint/api/infrastructure/vertx/persistence/postgres/PostgresContainerExtension.kt
package codes.fdk.blueprint.api.infrastructure.vertx.persistence.postgres

import org.junit.jupiter.api.extension.AfterEachCallback
import org.junit.jupiter.api.extension.BeforeAllCallback
import org.junit.jupiter.api.extension.ExtensionContext
import org.testcontainers.containers.PostgreSQLContainer

class PostgresContainerExtension : BeforeAllCallback, AfterEachCallback {

    companion object {
        val postgreSQLContainer = PostgreSQLContainer<Nothing>("postgres:13-alpine").apply {
            start()
        }
    }

    override fun beforeAll(context: ExtensionContext) {
        System.setProperty("PGHOST", postgreSQLContainer.containerIpAddress)
        System.setProperty("PGPORT", postgreSQLContainer.firstMappedPort.toString())
        System.setProperty("PGUSERNAME", postgreSQLContainer.username)
        System.setProperty("PGPASSWORD", postgreSQLContainer.password)
        System.setProperty("PGDATABASE", postgreSQLContainer.databaseName)
    }

    override fun afterEach(context: ExtensionContext?) {
        postgreSQLContainer.execInContainer(
            "psql",
            "-U",
            postgreSQLContainer.username,
            "-d",
            postgreSQLContainer.databaseName,
            "-c",
            "TRUNCATE TABLE categories;"
        )
    }


}